src/services/bill_manager/cancel_invoice.rs

```diff
@@ -0,0 +1,81 @@
+use crate::client::{Mpesa, MpesaResult};
+use crate::environment::ApiEnvironment;
+use crate::errors::MpesaError;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct CancelInvoicePayload<'mpesa> {
+    external_reference: &'mpesa str,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct CancelInvoiceResponse {
+    #[serde(rename(deserialize = "rescode"))]
+    pub res_code: String,
+    #[serde(rename(deserialize = "resmsg"))]
+    pub res_msg: String,
+    #[serde(rename(deserialize = "Status_Message"))]
+    pub status_message: String,
+}
+
+#[derive(Debug)]
+pub struct CancelInvoiceBuilder<'mpesa, Env: ApiEnvironment> {
+    client: &'mpesa Mpesa<Env>,
+    external_references: Vec<CancelInvoicePayload<'mpesa>>,
+}
+
+impl<'mpesa, Env: ApiEnvironment> CancelInvoiceBuilder<'mpesa, Env> {
+    /// Creates a new Bill Manager Cancel invoice builder
+    pub fn new(client: &'mpesa Mpesa<Env>) -> CancelInvoiceBuilder<'mpesa, Env> {
+        CancelInvoiceBuilder {
+            client,
+            external_references: vec![],
+        }
+    }
+
+    /// Adds `external_references`
+    pub fn external_references(
+        mut self,
+        external_references: Vec<&'mpesa str>,
+    ) -> CancelInvoiceBuilder<'mpesa, Env> {
+        self.external_references = external_references
+            .into_iter()
+            .map(|external_reference| CancelInvoicePayload { external_reference })
+            .collect();
+        self
+    }
```

Comment:
Is this method adding a reference? Adding multiple references?

the signature suggests otherwise though, looks like we are re-assigning existing references

Reply:
Ahh yes, it's reassigning references/string slices. I wanted the convenience of just passing a vector of external references (&str), rather than passing in a vector of `CancelInvoice { external_reference: "" }` structs. What would you recommend here?

Reply:
Ahh I see now, makes sense to pass the vec of strings, though this could also be another case for having a separate method to add a single reference to `self. external_references` perhaps. WDYT?

Reply:
Ahh you're right, will add it as a method as well 👍